util/restful/router.go

```
@@ -54,9 +58,49 @@ func (xlog *ExchangeLogger) Exchange() (data interface{}, status int) {
 	return
 }
 
+type loggingResponseWriter struct {
+	http.ResponseWriter
+	req          *http.Request
+	log          logging.LogSink
+	resourceName string
+	start        time.Time
+	statusCode   int
+}
+
+// WriteHeader implements and overrides ResponseWriter on loggingResponseWriter -
+// it records the status as reported for logging.
+func (lrw *loggingResponseWriter) WriteHeader(status int) {
+	lrw.statusCode = status
+	lrw.ResponseWriter.WriteHeader(status)
+}
+
+// Write logs the response. If ContentLength is set, we use that, otherwise we report a 0 length.
+// Unfortunately, ResponseWriter's contract makes it impossible to get the true content length.
+func (lrw loggingResponseWriter) Write(b []byte) (int, error) {
+	n, err := lrw.ResponseWriter.Write(b)
+
+	// ParseInt returns 0 and an syntax error if the provided string doesn't parse well.
```

Appreciate this comment for swallowing the error.